ppl/zqd/recruiter/recruiter_test.go

```diff
@@ -0,0 +1,293 @@
+package recruiter
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Register1(t *testing.T, addr string, nodename string, fp int, np int, np1 int, rp int) *WorkerPool {
+	wp := NewWorkerPool()
+	wp.Register(addr, nodename)
+	AssertPoolLen(t, wp, nodename, fp, np, np1, rp)
+	return wp
+}
+
+func AssertPoolLen(t *testing.T, wp *WorkerPool, name string, fp int, np int, np1 int, rp int) {
+	assert.Equal(t, fp, wp.LenFreePool(), "FreePool len=%d", fp)
+	require.Equal(t, np, wp.LenNodePool(), "NodePool len=%d", np)
+	assert.Equal(t, rp, wp.LenReservedPool(), "ReservedPool len=%d", rp)
+}
+
+func TestRegister1(t *testing.T) {
```

Comment:
Could you group the short tests here that verify bad or duplicate calls have no impact into a single top-level test (maybe "TestBadCalls")? I think even calling `t.Run` for them is probably overkill, and it will be clearer to have a single test case.

Reply:
OK. I will consolidate a few of the other simple tests too.